compiler/rustc_errors/src/styled_buffer.rs

```diff
@@ -13,34 +13,10 @@ impl StyledBuffer {
         StyledBuffer { text: vec![], styles: vec![] }
     }
 
-    fn replace_tabs(&mut self) {
-        for (line_pos, line) in self.text.iter_mut().enumerate() {
-            let mut tab_pos = vec![];
-            for (pos, c) in line.iter().enumerate() {
-                if *c == '\t' {
-                    tab_pos.push(pos);
-                }
-            }
-            // start with the tabs at the end of the line to replace them with 4 space chars
-            for pos in tab_pos.iter().rev() {
-                assert_eq!(line.remove(*pos), '\t');
-                // fix the position of the style to match up after replacing the tabs
-                let s = self.styles[line_pos].remove(*pos);
-                for _ in 0..4 {
-                    line.insert(*pos, ' ');
-                    self.styles[line_pos].insert(*pos, s);
-                }
```

Comment:
I'm slightly concerned by this removal. Have you tried to see the output in your local CLI? I need to double check the code, but can you confirm that the underlines are all colored correctly?

Reply:
I believe this is safe, since the style handling here is about adjusting them for the adding tabs, but we now handle the tabs earlier on, so there's no need to adjust at this stage any more.

The output looks correct to me as well. Here are a few examples:

<img width="839" alt="image" src="https://user-images.githubusercontent.com/279572/101510397-cab6d700-3971-11eb-856b-d4563aee4b9f.png">

<img width="837" alt="image" src="https://user-images.githubusercontent.com/279572/101510716-2e410480-3972-11eb-92e3-0746b86fbd54.png">